client-spark/common/src/main/java/org/apache/uniffle/shuffle/manager/RssShuffleManagerBase.java

```diff
@@ -664,6 +664,13 @@ public boolean reassignOnStageResubmit(
     String stageIdAndAttempt = stageId + "_" + stageAttemptNumber;
     RssStageInfo rssStageInfo =
         rssStageResubmitManager.recordAndGetServerAssignedInfo(shuffleId, stageIdAndAttempt);
+    if (rssStageInfo.isReassigned()) {
```

Comment:
Is this right? The process of marking reassigned should be guarded in the lock

Reply:
Is it possible to reset to false after being marked as Reassigned?